shared_utils.py

```diff
@@ -117,7 +119,42 @@ def check_styles(soup: bs4.BeautifulSoup) -> None:
         warn('style_no_intro')
     if not get_elem_containing_text(soup, 'h1', 'references'):
         warn('style_no_refs')
-
+    # Check every numbered reference appears in the text in square brackets
+    dom = etree.HTML(str(soup))
+    ref_lis = dom.xpath("//h1[contains(text(), 'REFERENCES')]/following-sibling::ol/li | //h1[.//*[contains(text(), 'REFERENCES')]]/following-sibling::ol/li") 
+    ref_in_ref = set( range(1,len(ref_lis)+1)  ) 
+    ref_in_text_raw = re.findall('\[([\d, ]+)\]', soup.get_text( strip=True) )
+    ref_in_text = set( [ int(i) for i in re.split(r'\D+',','.join(ref_in_text_raw)) ] )
+    mismatched_ref = ref_in_ref.symmetric_difference(ref_in_text)
+    if len(mismatched_ref) > 0:
+        warn('mismatched_refs', sorted (mismatched_ref) )
+    # Check references are complete; executes anystyle in shell
+    refs = [ ' '.join( "".join(ref_li.itertext() ).split() ) for ref_li in ref_lis ] #lxml makes this ugly
+    fname = os.path.join(output_dir, "extracted_refs.txt")
+    with open(os.path.join(fname), 'w') as ofile:
+        for ref in refs:
+            ofile.write(ref + '\n')
+    subprocess.call([CONFIG['anystyle_path'], '-f' , 'json', 'parse',
```

Comment:
Could you add `--overwrite` to this? Otherwise it doesn't seem to work when re-running a conversion task.

Reply:
Good call, I've been deleting b/w runs